ssl/ssl.go

```diff
@@ -265,18 +321,80 @@ func createCert(certInfo map[string]CertInfo, namespace, clusterDomain string, s
 		return nil, fmt.Errorf("Failed to sign cert: %s", err)
 	}
 
-	if len(info.PrivateKeyName) == 0 {
-		return nil, fmt.Errorf("Certificate %s created with empty private key name", id)
+	return &info, nil
+}
+func createKubeCSR(csri util.CertificateSigningRequestInterface, request []byte, info CertInfo, namespace, id string, autoApproval bool) (*CertInfo, error) {
+	csr := &certificates.CertificateSigningRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: util.ConvertNameToKey(fmt.Sprintf("%s-%s", namespace, id)),
+		},
+		Spec: certificates.CertificateSigningRequestSpec{
+			Request: request,
+			Usages: []certificates.KeyUsage{
+				certificates.UsageClientAuth,
+				certificates.UsageServerAuth,
+			},
+			// expiration is determined by apiserver and cannot be configured
```

Comment:
Do we need to figure out the procedure to renew the certs? (Probably as a new task)

Reply:
The procedure should be the same as for any other certs we generate: secrets rotation.

I would hope that maybe Stratos will eventually have a way to warn an admin about certs that are going to expire soon. Alternatively the admin could run something like `havener` to quickly get the current state.

Reply:
Okay, we _do_ have it [documented](https://susedoc.github.io/doc-cap/develop/cap-guides/html/cha-cap-certificates.html#sec-cap-secrets-rotation), so we should be fine.